src/main/java/com/mathworks/ci/RunMatlabBuildBuilder.java

```diff
@@ -0,0 +1,152 @@
+package com.mathworks.ci;
+
+/**
+ * Copyright 2022 The MathWorks, Inc.
+ *  
+ */
+
+import java.io.IOException;
+import javax.annotation.Nonnull;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+import org.kohsuke.stapler.StaplerRequest;
+import hudson.EnvVars;
+import hudson.Extension;
+import hudson.FilePath;
+import hudson.Launcher;
+import hudson.Launcher.ProcStarter;
+import hudson.model.AbstractProject;
+import hudson.model.Result;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.model.Computer;
+import hudson.tasks.BuildStepDescriptor;
+import hudson.tasks.Builder;
+import jenkins.tasks.SimpleBuildStep;
+import net.sf.json.JSONObject;
+
+public class RunMatlabBuildBuilder extends Builder implements SimpleBuildStep, MatlabBuild {
+    private int buildResult;
+    private String tasks;
+
+    @DataBoundConstructor
+    public RunMatlabBuildBuilder() {}
+
+    // Getter and Setters to access local members
+    @DataBoundSetter
+    public void setTasks(String tasks) {
+        this.tasks = tasks;
+    }
+
+    public String getTasks() {
+        return this.tasks;
+    }
+    
+    @Extension
+    public static class RunMatlabBuildDescriptor extends BuildStepDescriptor<Builder> {
+
+        // Overridden Method used to show the text under build dropdown
+        @Override
+        public String getDisplayName() {
+            return Message.getValue("Builder.build.builder.display.name");
+        }
+
+        @Override
+        public boolean configure(StaplerRequest req, JSONObject formData) throws FormException {
+            save();
+            return super.configure(req, formData);
+        }
+
+        /*
+         * This is to identify which project type in jenkins this should be applicable.(non-Javadoc)
+         * 
+         * @see hudson.tasks.BuildStepDescriptor#isApplicable(java.lang.Class)
+         * 
+         * if it returns true then this build step will be applicable for all project type.
+         */
+        @Override
+        public boolean isApplicable(
+                @SuppressWarnings("rawtypes") Class<? extends AbstractProject> jobtype) {
+            return true;
+        }
+    }
+
+    @Override
+    public void perform(@Nonnull Run<?, ?> build, @Nonnull FilePath workspace,
+            @Nonnull Launcher launcher, @Nonnull TaskListener listener)
+            throws InterruptedException, IOException {
+
+        // Get the environment variable specific to the this build
+        final EnvVars env = build.getEnvironment(listener);
+
+        // Invoke MATLAB build and transfer output to standard
+        // Output Console
+
+
+        buildResult = execMatlabCommand(workspace, launcher, listener, env);
+
+        if (buildResult != 0) {
+            build.setResult(Result.FAILURE);
+        }
+    }
+
+    private int execMatlabCommand(FilePath workspace, Launcher launcher,
```

Comment:
Lots of duplication in the `execMatlabCommand` and `createMatlabScriptByName` methods in all 6 of the Builder and StepExecution files. Something to consider going forward

Reply:
+1